tests/RobotFramework/tests/tedge/diag/predefined_plugins.robot

```diff
@@ -0,0 +1,71 @@
+*** Settings ***
+Library             String
+Library             ThinEdgeIO
+
+Suite Setup         Custom Suite Setup
+Suite Teardown      Get Suite Logs
+
+Test Tags           theme:troubleshooting    theme:cli    theme:plugins
+
+
+*** Test Cases ***
+01_tedge
+    ${log_names}=    Create List
+    ...    output.log
+    ...    tedge-agent.log
+    ...    tedge-mapper-c8y.log
+    ...    tedge-mapper-collectd.log
+    ...    tedge-config-list.log
+    ...    tedge.toml
+    FOR    ${log_name}    IN    @{log_names}
+        File Size Is Not Zero    ${log_name}
+    END
+    Execute Command    diff /etc/tedge/tedge.toml /results/test/01_tedge/tedge.toml
+
+02_os
+    File Size Is Not Zero    output.log
+
+03_mqtt
+    ${log_names}=    Create List
+    ...    output.log
+    ...    tedge-mqtt-sub.log
+    ...    tedge-mqtt-sub-retained-only.log
+    FOR    ${log_name}    IN    @{log_names}
+        File Size Is Not Zero    ${log_name}
+    END
+
+04_workflow
+    File Size Is Not Zero    output.log
+    Execute Command    diff -qr /var/log/tedge/agent/workflow-* /results/test/04_workflow/workflow-*
+
+05_entities
+    File Size Is Not Zero    output.log
+
+06_internal
+    ${log_names}=    Create List
+    ...    output.log
+    ...    entity_store.jsonl
+    FOR    ${log_name}    IN    @{log_names}
+        File Size Is Not Zero    ${log_name}
+    END
+    Execute Command    diff /etc/tedge/.agent/entity_store.jsonl /results/test/06_internal/entity_store.jsonl
+
+07_mosquitto
+    ${log_names}=    Create List
+    ...    output.log
+    ...    mosquitto-journal.log
+    FOR    ${log_name}    IN    @{log_names}
+        File Size Is Not Zero    ${log_name}
+    END
+
+
+*** Keywords ***
+File Size Is Not Zero
+    [Arguments]    ${filename}
+    Execute Command    test -s /results/test/${TEST NAME}/${filename}
+
+Custom Suite Setup
+    Setup
+    Execute Command    mkdir -p /results
+    Start Service    tedge-mapper-collectd
```

Comment:
Is there possibly a race here? Are the tests assuming the diagnostic collection happens after `tedge-mapper-collectd` publishes something?